core/modules/pluginswitcher.js

```diff
@@ -47,16 +47,40 @@ PluginSwitcher.prototype.switchPlugins = function() {
 	}
 	// Accumulate the titles of the plugins that we need to load
 	var plugins = [],
+		unregisteredTiddlers = [],
+		registeredTiddlers = [],
 		self = this,
+		accumulateLanguageAddOn = function(title) {
```

Comment:
We use the pluginswitcher class for both themes and language switching. Can't we keep things generic so that we get theme addons for free?

Reply:
You are right: I also don't like this part of the new code too much. 

--- 

Themes and languages are technically  plugins, BUT the internal handling is slightly different. ... Those differences cause problems here. 

A language-add on will probably need different handling than themes. .. Themes are closer to plugin handling.